clippy_lints/src/missing_const_for_fn.rs

```diff
@@ -114,6 +115,18 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for MissingConstForFn {
     }
 }
 
+fn is_trait_method(cx: &LateContext<'_, '_>, hir_id: HirId) -> bool {
```

Comment:
I think you may be able to use https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/fn.trait_of_item.html instead of this function. There's no docs on the function, so it might have panic cases, but as long as it's only used on methods, it should be fine

Reply:
I tried that and it only reported the traits of the default impl methods, not of normal trait methods like in the example :/